### DISCUSSION TITLE
Open source Desktop Agent? (2.1)

### DISCUSSION THREAD

**Discussion by jfbaro** [finos/FDC3#1356] [General]
Hello,
We are exploring the implementation of FDC3 to enable communication between our custom desktop applications (Windows .NET-based), custom web-based applications (typically Angular), and off-the-shelf solutions like Excel, Salesforce, and other SaaS platforms we use.
We're considering using FDC3-Sail as our desktop agent. Could you please advise whether we can use FDC3-Sail as-is and achieve all the benefits that paid solutions like Glue42 offer?
Our main objectives are to implement the core features of FDC3:
1. **Unified Desktop Environment:** Run desktop and web-based applications within a single "desktop" environment, primarily on Windows laptops (no MacOS or Linux support needed).
2. **Two-Way Communication:** Enable seamless two-way communication between applications.
3. **Automatic Application Launch:** Automatically launch desktop or web applications when an intent is triggered.
4. **Workspace Persistence:** Save the "desktop" layout, including screen sizes and other properties, so it can be restored in the same state the next day.
5. **App Directory Management:** Utilize an App Directory to manage applications that respond to intents.
6. **Administrative Configuration:** Provide an admin interface where we can configure desktop agents for different user types or roles (e.g., limiting intents for particular users). This admin should integrate with our Single Sign-On (SSO) system so administrators can use their normal internal credentials.
7. **Security and Compliance:** Ensure privacy, compliance, and SSO access to meet typical financial institution audit and regulatory requirements.
8. **Production Deployment:** Deploy this solution in a production environment for internal users only—not for our customers or brokers.
We would greatly appreciate any guidance or insights you can provide.
Thank you!

**Comment by kriswest** [comment]
Hi, I'm not a sail user (and in fact work for interop.io who publish io.connect desktop/browser), but AFAIK Sail does not have an adaptor for .NET applications to connect to the DA, nor the ability to save workspace layouts or a variety of other features you mention (at least as far as I know). 

It does have an FDC3 2.0 implementation and support for an app directory however.

**Comment by robmoffat** [comment]
Hi @jfbaro,

I've picked up Sail maintenance and development recently.  I am working on Sail 2.0 conformance and Sail using FDC3 on the Web right now.  Perhaps it would be good to meet up and discuss your shopping list?

My calendar link is here: https://calendly.com/rob-moffat/one-to-one-meeting

**Reply by jfbaro** [reply]
Thanks @robmoffat. This is great! Let me first align that list with the stakeholders, and I will sure get back to you. 

Appreciated!
